src/Plugin/Field/FieldFormatter/InstagramEmbedFormatter.php

```diff
@@ -19,13 +20,35 @@
  *   id = "instagram_embed",
  *   label = @Translation("Instagram embed"),
  *   field_types = {
- *     "string_long"
+ *     "link", "string", "string_long"
  *   }
  * )
  */
 class InstagramEmbedFormatter extends FormatterBase {
 
   /**
+   * Extracts the embed code from a field item.
+   *
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *   The field item.
+   *
+   * @return string|null
+   *   The embed code, or NULL if the field type is not supported.
+   */
+  protected function getEmbedCode(FieldItemInterface $item) {
```

Comment:
You don't need to implement this function. Just use EmbedCodeValueTrait and it will bring it for you.